lib/remote-connector.js

```diff
@@ -91,14 +129,56 @@ function createProxyMethod(Model, remotes, remoteMethod) {
       callback = args.pop();
     }
 
+    // Take care not to overwrite method args
+    var numberOfDefaultArgs = remoteMethod.accepts.length - headerArgs.length;
+    headerArgs.forEach(function(headerArg, idx) {
+      args[idx + numberOfDefaultArgs] = headerArg;
+    });
+
+    if (auth && auth.accessTokenId) {
+      // Find correct index for auth header
+      remoteMethod.accepts.forEach(function(accept, idx) {
+        if (accept.name === 'authorization') {
+          args[idx] = auth.accessTokenId;
+        }
+      });
+    }
+
+    // Make sure gaps in array are numbered.
+    for (var i = 0, argsLength = args.length; i < argsLength; i++) {
+      args[i] = args[i] || void 0;
+    }
+
     if (remoteMethod.isStatic) {
       return remotes.invoke(remoteMethod.stringName, args, callback);
     }
 
     var ctorArgs = [this.id];
     return remotes.invoke(remoteMethod.stringName, ctorArgs, args, callback);
-  }
+  };
 }
 
-function noop() {
+function interceptUserRemotes(remotes, auth) {
+  remotes.after('User.logout', function(ctx, next) {
+    auth.clearUser();
+    auth.clearStorage();
+    next();
+  });
+
+  remotes.before('User.login', function(ctx, next) {
+    auth.rememberMe = ctx.req.body.rememberMe || false;
```

Comment:
@bajtos Not sure how to go about this. Should this information be available right on the context object? i.e. ctx.rememberMe? This will require changes to https://github.com/strongloop/strong-remoting/blob/master/lib/rest-adapter.js#L117 to add this property right on ctx.